src/inversion_scripts/operators/operator_utilities.py

```diff
@@ -52,15 +78,119 @@ def read_geoschem(date, gc_cache, build_jacobian=False, sensi_cache=None):
+            else:
+                pert_simulations_dict[run_num].append(sv_elem)
+                
+                
+        gc_date = pd.to_datetime(date, format='%Y%m%d_%H')
+        ds_all = [concat_tracers(k, gc_date, config, v) for k,v in pert_simulations_dict.items()]
+        ds_sensi = xr.concat(ds_all, 'element')
+        ds_sensi.load()
+        
+        sensitivities = ds_sensi["ch4"].values
         # Reshape so the data have dimensions (lon, lat, lev, grid_element)
         sensitivities = np.einsum("klji->ijlk", sensitivities)
-        sensitivity_data.close()
-        dat["Sensitivities"] = sensitivities
+        dat["jacobian_ch4"] = sensitivities
+        
+        # get emis base, which is also BC base
+        ds_emis_base = concat_tracers('0001', gc_date, config, [0], baserun=True)
+        ds_emis_base.load()
+        dat['emis_base_ch4'] = np.einsum('klji->ijlk', ds_emis_base['ch4'].values)
+        
+        # get OH base, run RunName_0000
+        # it's always here whether OptimizeOH is true or not
+        # so we can keep it here for convenience
+        ds_oh_base = concat_tracers('0000', gc_date, config, [0])
+        ds_oh_base.load()
+        dat['oh_base_ch4'] = np.einsum('klji->ijlk', ds_oh_base['ch4'].values)
+        
 
     return dat
 
+
+def concat_tracers(run_id, gc_date, config, sv_elems, baserun=False):
+    """
+    Concatenate CH4 tracers from all jacobian GEOS-Chem simulations.
+    Tracers are assigned a new dimension: "element"
+
+    Arguments
+        run_id   [str]         : ID for Jacobian GEOS-Chem run, e.g. "0001"
+        gc_date  [pd.Datetime] : date object, specifies Ymd_h
+        config   [dict]        : dictionary of IMI config file
+        sv_elems [list]        : list of state vector element tracers in this simulations
+        baserun  [bool]        : If True, only the base variable in the simulation will
+                                 be opened, and the function will just return this one
+                                 variable instead of concatenating all elements. Used to 
+                                 get the base for calculating the sensitivities.
+
+    Returns
+        ds_concat [xarray.Dataset] : dataset of all Jacobian CH4 at this timestep for 
+                                     all tracer runs. Has dimensions
+                                        - lat
+                                        - lon
+                                        - lev
+                                        - element
+                                    
+
+    """
+    prefix = config['OutputPath'] + '/' + config['RunName'] + '/jacobian_runs'
+    j_dir = f"{prefix}/{config['RunName']}_{run_id}/OutputDir"
+    file_stub = gc_date.strftime('GEOSChem.SpeciesConc.%Y%m%d_0000z.nc4')
+    dsmf = xr.open_dataset(
+        '/'.join([j_dir,file_stub]),
+        chunks = 'auto'
+    )
+    keepvars = [f'SpeciesConcVV_CH4_{i:04}' for i in sv_elems]
+    if len(keepvars) == 1:
```

Comment:
Okay, I think I understand this function now. Could we change this conditional to check if is_bc_element == True or is_OH_element == True: because I think this would break if the number of tracers in a perturbation simulation was set to 1.

Let me know if this isn't true.

Reply:
Yes, great catch. If the number of tracers is set to 1, does the variable name have `_0001` appended?

Reply:
Added a check for OH or BC elements